src/main/java/seedu/address/logic/parser/MarkPaidCommandParser.java

```diff
@@ -24,22 +25,32 @@ public class MarkPaidCommandParser implements Parser<MarkPaidCommand> {
      */
     public MarkPaidCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_MONTHPAID);
+        String preamble = argMultimap.getPreamble().trim();
+        Set<MonthPaid> monthsPaid = parseMonthsPaidForMarkPaid(argMultimap.getAllValues(PREFIX_MONTHPAID));
 
-        if (argMultimap.getPreamble().isEmpty() || !ParserUtil.isValidIndex(argMultimap.getPreamble())) {
+        if (preamble.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkPaidCommand.MESSAGE_USAGE));
         }
 
-        Index index;
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+        MarkPaidTarget target;
+        if (preamble.equalsIgnoreCase("all")) {
```

Comment:
Not sure if ignore case is a good idea here. Not suggesting to change it, but I suggest u open an issue about this so u don't forget about it